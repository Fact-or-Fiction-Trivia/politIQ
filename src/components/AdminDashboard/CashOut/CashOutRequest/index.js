import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CashOutRequest extends Component {
  constructor(props) {
    super(props);
    this.state = {
      num: 0,
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.cashoutData !== this.props.cashoutData) {
      this.updateNum(nextProps.cashoutData);
      return true;
    }
    if (nextState !== this.state) {
      return true;
    }
    return false;
  }

  updateNum = (data) => {
    if (data === null || data === undefined) return;
    const num = Object.keys(data).length;
    this.setState({ num });
  }

  render() {
    return (
      <div
        className="reviewWidget cashOutWidget"
        onClick={this.props.toggleCashOut}
        onKeyPress={(event) => {
          if (event.keyCode === 13) {
            this.props.toggleCashOut();
          }
        }}
        role="button" tabIndex={0}
      >
        <div className="counterDisplay">{this.state.num}</div>
        <h1>Cash Out Requests</h1>
      </div>
    );
  }
}

CashOutRequest.propTypes = {
  cashoutData: PropTypes.object,
  toggleCashOut: PropTypes.func,
};

export default CashOutRequest;
