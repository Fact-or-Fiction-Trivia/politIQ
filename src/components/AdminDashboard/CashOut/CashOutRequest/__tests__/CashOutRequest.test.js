import React from 'react';
import { shallow, mount } from 'enzyme';

import CashOutRequest from '../index';

const toggleCashout = jest.fn();

const fakeCashoutData = {
  GvM7fEkYp4fcaSuScwixV0xinfT2: {
    date: '2019-08-05',
    displayName: 'Hannah',
    email: 'hannah.werman@gmail.com',
    moneyEarned: 5,
  },
  GvM7fEkYp4fcaSuScwixV0xinfT: {
    date: '2019-08-05',
    displayName: 'Hannah',
    email: 'hannah.werman@gmail.com',
    moneyEarned: 5,
  },
};

describe('<CashOutRequest />', () => {
  it('should render correctly and match snapshot', () => {
    const wrapper = shallow(
      <CashOutRequest toggleCashout={toggleCashout} cashoutData={fakeCashoutData} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should display the correct number of cashout requests', async () => {
    jest.spyOn(CashOutRequest.prototype, 'shouldComponentUpdate');
    const wrapper = mount(<CashOutRequest toggleCashout={toggleCashout} cashoutData={null} />);

    wrapper.setProps({ cashoutData: fakeCashoutData });
    expect(CashOutRequest.prototype.shouldComponentUpdate.mock.calls.length).toBe(2);

    const counterDisplay = wrapper.find('div.counterDisplay');
    const cashoutReqs = Object.keys(wrapper.props().cashoutData).length;
    expect(parseInt(counterDisplay.text(), 10)).toEqual(cashoutReqs);
  });
});
