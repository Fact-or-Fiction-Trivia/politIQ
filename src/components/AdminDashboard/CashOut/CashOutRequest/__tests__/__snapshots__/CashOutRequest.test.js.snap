// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CashOutRequest /> should render correctly and match snapshot 1`] = `
<div
  className="reviewWidget cashOutWidget"
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="counterDisplay"
  >
    0
  </div>
  <h1>
    Cash Out Requests
  </h1>
</div>
`;
